src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -6711,6 +6762,67 @@ components:
           items:
             type: string
 
+    SnapshotAccessRequest:
+      type: object
+      properties:
+        name:
+          type: string
+        researchPurposeStatement:
+          type: string
+        datasetRequest:
+          $ref: '#/components/schemas/SnapshotRequest'
+      description: Request for access to a snapshot.
+
+    SnapshotRequest:
+      type: object
+      properties:
+        cohorts:
+          type: array
+          items:
+            $ref: '#/components/schemas/Cohort'
+        conceptSets:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderDatasetConceptSets'
+        valueSets:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderFeatureValueGroup'
+      description: Specification of requested a snapshot, returned in a SnapshotAccessRequest.
+
+    Cohort:
+      type: object
+      properties:
+        name:
+          type: string
+        criteriaGroups:
+          type: array
+          items:
+            $ref: '#/components/schemas/CriteriaGroup'
```

Comment:
Same as above, I would personally prefix this with `SnapshotBuilder` for searchability and clarity.